self-tests.yml

```diff
@@ -890,8 +893,30 @@ buildvariants:
     tasks:
       - name: ".smoke"
 
+  - name: lint-markdown
+    display_name: Lint Markdown
+    expansions:
+      goos: linux
+      goarch: amd64
+      GOROOT: /opt/golang/go1.24
+      mongodb_url: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2204-8.0.0.tgz
+      mongosh_url: https://downloads.mongodb.com/compass/mongosh-2.0.2-linux-x64.tgz
+      notary_client_url: https://macos-notary-1628249594.s3.amazonaws.com/releases/client/v3.1.2/linux_amd64.zip
```

Comment:
I believe only GOROOT is necessary for lint-markdown. Can we please test it without the rest and remove the ones that are not needed?